java/clients/src/main/java/sleeper/clients/api/SleeperClient.java

```diff
@@ -98,19 +98,32 @@ private SleeperClient(Builder builder) {
      * @return            the client
      */
     public static SleeperClient createForInstanceId(String instanceId) {
+        return createForInstanceId(instanceId, Executors.newFixedThreadPool(10));
+    }
+
+    /**
+     * Creates a client to interact with the instance of Sleeper with the given ID.
+     * Will use the default AWS configuration.
+     *
+     * @param  instanceId      the instance ID
+     * @param  executorService an executor service
+     * @return                 the client
+     */
+    public static SleeperClient createForInstanceId(String instanceId, ExecutorService executorService) {
```

Comment:
I'm not sure I agree with the idea of exposing the implementation in this way. We have the builder as a way to initialise the client yourself. Having too many static constructors seems to make it harder than it already is to customise that in the way you actually want to, since there's no guarantee this method signature will be what you want.

Maybe we should add a shutdown or close method to the client instead? Maybe we should discuss this on the issue.